Dockerfile-alpine.template

```diff
@@ -17,7 +17,7 @@ ENV LANG en_US.utf8
 RUN mkdir /docker-entrypoint-initdb.d
 
 ENV PG_MAJOR {{ env.version }}
-ENV PG_VERSION {{ .version }}
+ENV PG_VERSION {{ .version | gsub("-"; "") }}
```

Comment:
I'm confused why we're doing all this footwork to convert `.version` into `14-beta1` when the canonical upstream version number is `14beta1` (and pretty much everywhere we use it we're stripping the `-` anyhow) -- what breaks when we don't?

Reply:
If we don't have a `-` then we get no tags:

https://github.com/docker-library/postgres/blob/0d2e407c7c9baf10e05a01811d9938f45c8cb40e/generate-stackbrew-library.sh#L89-L92

Reply:
Right, we get _just_ `14beta1` as a tag, right?  That's probably a good thing to help further discourage users from using these accidentally?

Reply:
No, we get zero tags.

```console
$ ./generate-stackbrew-library.sh 14
# this file is generated via https://github.com/docker-library/postgres/blob/517c64f87e6661366b415df3f2273c76cea428b0/generate-stackbrew-library.sh

Maintainers: Tianon Gravi <admwiggin@gmail.com> (@tianon),
             Joseph Ferguson <yosifkit@gmail.com> (@yosifkit)
GitRepo: https://github.com/docker-library/postgres.git

Tags: 
Architectures: amd64, arm32v5, arm32v7, arm64v8, i386, mips64le, ppc64le, s390x
GitCommit: ebca4742f7140afb3d20978ae74d21bcc99d172b
Directory: 14/buster

Tags: 
Architectures: amd64, arm32v6, arm32v7, arm64v8, i386, ppc64le, s390x
GitCommit: ebca4742f7140afb3d20978ae74d21bcc99d172b
Directory: 14/alpine
```

```console
$ bash -x ./generate-stackbrew-library.sh 14
...
+ fullVersion=14beta2
+ origVersion=14beta2
+ versionAliases=()
+ '[' 14beta2 '!=' 14 -a 14beta2 '!=' 14beta2 ']'
+ case "$origVersion" in
+ versionAliases+=(${aliases[$version]:-})
```

Reply:
Ahh, we need to axe the "special `case`" below that:

```diff
diff --git a/generate-stackbrew-library.sh b/generate-stackbrew-library.sh
index b831667..6b968ed 100755
--- a/generate-stackbrew-library.sh
+++ b/generate-stackbrew-library.sh
@@ -90,14 +90,8 @@ for version; do
 		versionAliases+=( $fullVersion )
 		fullVersion="${fullVersion%[.-]*}"
 	done
-	# skip unadorned "version" on prereleases: https://www.postgresql.org/developer/beta/
-	# - https://github.com/docker-library/postgres/issues/662
-	# - https://github.com/docker-library/postgres/issues/784
-	case "$origVersion" in
-		*alpha* | *beta* | *rc*) ;;
-		*) versionAliases+=( $version ) ;;
-	esac
 	versionAliases+=(
+		$version
 		${aliases[$version]:-}
 	)
 
```

Reply:
That makes it bad:
```console
$ ./generate-stackbrew-library.sh 14
# this file is generated via https://github.com/docker-library/postgres/blob/517c64f87e6661366b415df3f2273c76cea428b0/generate-stackbrew-library.sh

Maintainers: Tianon Gravi <admwiggin@gmail.com> (@tianon),
             Joseph Ferguson <yosifkit@gmail.com> (@yosifkit)
GitRepo: https://github.com/docker-library/postgres.git

Tags: 14, 14-buster
Architectures: amd64, arm32v5, arm32v7, arm64v8, i386, mips64le, ppc64le, s390x
GitCommit: ebca4742f7140afb3d20978ae74d21bcc99d172b
Directory: 14/buster

Tags: 14-alpine
Architectures: amd64, arm32v6, arm32v7, arm64v8, i386, ppc64le, s390x
GitCommit: ebca4742f7140afb3d20978ae74d21bcc99d172b
Directory: 14/alpine
```

<details>
<summary>Diff:</summary>

```diff
diff --git a/generate-stackbrew-library.sh b/generate-stackbrew-library.sh
index 4ecfc95..070c25e 100755
--- a/generate-stackbrew-library.sh
+++ b/generate-stackbrew-library.sh
@@ -93,11 +93,12 @@ for version; do
        # skip unadorned "version" on prereleases: https://www.postgresql.org/developer/beta/
        # - https://github.com/docker-library/postgres/issues/662
        # - https://github.com/docker-library/postgres/issues/784
-       case "$origVersion" in
-               *alpha* | *beta* | *rc*) ;;
-               *) versionAliases+=( $version ) ;;
-       esac
+#      case "$origVersion" in
+#              *alpha* | *beta* | *rc*) ;;
+#              *) versionAliases+=( $version ) ;;
+#      esac
        versionAliases+=(
+               $version
                ${aliases[$version]:-}
        )
 
diff --git a/versions.json b/versions.json
index 7b3c96c..b919cc0 100644
--- a/versions.json
+++ b/versions.json
@@ -110,7 +110,7 @@
     ],
     "major": 14,
     "sha256": "ffe64a76f50a2363443c1c9dc2195138933e931e351b74fb35a7935eae7c60a5",
-    "version": "14-beta2"
+    "version": "14beta2"
   },
   "9.6": {
     "alpine": "3.14",
```

</details>